### PR TITLE
development: microservices mode docker compose scripts add compatibility for newer compose and address shellcheck linting

### PR DESCRIPTION
**Pull request by francoposa**
#### What this PR does

Adds bash fn to call 'docker compose' if 'docker-compose' is not available.
This pattern could be flipped, trying to call 'docker compose' first if it's available, since that is the new one. But I think most users are on systems still with `docker-compose`, plus the order doesn't _really_ matter.

Also address shellcheck warnings, adding quoting around shell variable expansion. 

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by 56quarters — APPROVED** [review]
Thanks!

**Comment by 56quarters** [comment]
How do you feel about updating the other compose setups?

**Comment by francoposa** [comment]
> How do you feel about updating the other compose setups?

can do, will push up today

**Comment by francoposa** [comment]
Now updated to expand this to the other compose scripts in the development dir, and tag `consul:1.15` since consul is not publishing a `latest` tag to docker hub anymore

**Review by 56quarters — APPROVED** [review]

File: development/mimir-microservices-mode/compose-up.sh [hunk 1]
```diff
@@ -6,13 +6,22 @@
...
+cd "$SCRIPT_DIR" && make
 
 # -gcflags "all=-N -l" disables optimizations that allow for better run with combination with Delve debugger.
 # GOARCH is not changed.
-CGO_ENABLED=0 GOOS=linux go build -mod=vendor -gcflags "all=-N -l" -o ${SCRIPT_DIR}/mimir ${SCRIPT_DIR}/../../cmd/mimir
-docker-compose -f ${SCRIPT_DIR}/docker-compose.yml build --build-arg BUILD_IMAGE=${BUILD_IMAGE} distributor-1
-docker-compose -f ${SCRIPT_DIR}/docker-compose.yml up $@
+CGO_ENABLED=0 GOOS=linux go build -mod=vendor -gcflags "all=-N -l" -o "${SCRIPT_DIR}"/mimir "${SCRIPT_DIR}"/../../cmd/mimir
+docker_compose -f "${SCRIPT_DIR}"/docker-compose.yml build --build-arg BUILD_IMAGE="${BUILD_IMAGE}" distributor-1
+docker_compose -f "${SCRIPT_DIR}"/docker-compose.yml up "$@"
```
Comment (pstibrany) [inline development/mimir-microservices-mode/compose-up.sh:27]: I don't think quotes around `$@` are correct -- that will convert all remaining argument into single argument.

Comment (francoposa) [reply development/mimir-microservices-mode/compose-up.sh:27]: So it does work with multiple args, e.g.

`./compose-up.sh -d minio memcached distributor-1 distributor-2 ingester-1 ingester-2 store-gateway-1 store-gateway-2 querier query-scheduler query-frontend nginx grafana`

The quoting suppresses this warning, which is about using globbing  
https://github.com/koalaman/shellcheck/wiki/SC2068:
```
Let's say you have four arguments: baz, foo bar, * and /*/*/*/*

"$@" will expand into exactly that: baz, foo bar, * and /*/*/*/*

$@ will expand into multiple other arguments: baz, foo, bar, file.txt, otherfile.jpg, and (eventually) a list of most files on the system
```

I could see that someone may want to use globbing with the docker-compose command?
It's probably fine to leave it unquoted, as it's not intended to be a live install script or something run on a production system.

Comment (pstibrany) [reply development/mimir-microservices-mode/compose-up.sh:27]: Let's keep as-is, until someone runs into a problem here.

**Pull request merged**